modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -1321,16 +1283,9 @@ private void deleteLocks(CommitData cd, final long commitTs) {
       mutations.add(m);
     }
 
-
-    ListenableFuture<Void> future =
+    CompletableFuture<Void> cfuture =
         env.getSharedResources().getBatchWriter().writeMutationsAsyncFuture(mutations);
-    Futures.addCallback(future, new CommitCallback<Void>(cd) {
-      @Override
-      protected void onSuccess(CommitData cd, Void v) throws Exception {
-        finishCommit(cd, commitTs);
-      }
-    }, env.getSharedResources().getAsyncCommitExecutor());
-
+    addCallback(cfuture, cd, (cd2, result) -> finishCommit(cd, commitTs));
```

Comment:
I noticed that finishCommit is not passed `cd2`.  Of course it still works fine because, `cd` and `cd2` are the same.  This made me think it may simplify the code to remove CommitData param from `OnSuccessInterface.onSuccess()`.  Then this code could be written as :

```java
addCallback(cfuture, cd, result -> finishCommit(cd, commitTs));
```